src/otx/core/model/detection_3d.py

```diff
@@ -295,3 +216,349 @@ def get_classification_layers(self, prefix: str = "model.") -> dict[str, dict[st
                 num_extra_classes = 6 * sample_model_dim - 5 * incremental_model_dim
                 classification_layers[prefix + key] = {"stride": stride, "num_extra_classes": num_extra_classes}
         return classification_layers
+
+
+class MonoDETRModel(ImageModel):
+    """A wrapper for MonoDETR 3d object detection model."""
+
+    __model__ = "mono_3d_det"
+
+    def __init__(self, inference_adapter: InferenceAdapter, configuration: dict[str, Any], preload: bool = False):
+        """Initializes a 3d detection model.
+
+        Args:
+            inference_adapter (InferenceAdapter): inference adapter containing the underlying model.
+            configuration (dict, optional): configuration overrides the model parameters (see parameters() method).
+              Defaults to dict().
+            preload (bool, optional): forces inference adapter to load the model. Defaults to False.
+        """
+        super().__init__(inference_adapter, configuration, preload)
+        self._check_io_number(3, 5)
+
+    def preprocess(self, inputs: dict[str, np.ndarray]) -> tuple[dict[str, Any], ...]:
+        """Preprocesses the input data for the model.
+
+        Args:
+            inputs (dict[str, np.ndarray]): a dict with image, calibration matrix, and image size
+
+        Returns:
+            tuple[dict[str, Any], ...]: a tuple with the preprocessed inputs and meta information
+        """
+        return {
+            self.image_blob_name: inputs["image"][None],
+            "calib_matrix": inputs["calib"],
+            "img_sizes": inputs["img_size"][None],
+        }, {
+            "original_shape": inputs["image"].shape,
+            "resized_shape": (self.w, self.h, self.c),
+        }
+
+    def _get_inputs(self) -> tuple[list[Any], list[Any]]:
```

Comment:
This functions looks almost same as `ImageModel`'s. Is there any reason to override the function?

Reply:
The original get_inputs allows only 2D and 4D inputs, which is not the case for this model